src/cargo/core/resolver/mod.rs

```diff
@@ -675,10 +709,16 @@ impl<'a> RegistryQueryer<'a> {
             candidate.replace = replace;
         }
 
-        // When we attempt versions for a package, we'll want to start at
-        // the maximum version and work our way down.
+        // When we attempt versions for a package we'll want to do so in a
+        // sorted fashion to pick the "best candidates" first. Currently we try
+        // prioritized summaries (those in `try_to_use`) and failing that we
+        // list everything from the maximum version to the lowest version.
         ret.sort_unstable_by(|a, b| {
-            b.summary.version().cmp(a.summary.version())
+            let a_in_previous = self.try_to_use.contains(a.summary.package_id());
+            let b_in_previous = self.try_to_use.contains(b.summary.package_id());
```

Comment:
Hm, maybe let’s change try_to_use to hash set to have better O?

Reply:
Ah it is indeed here! The argument coming in is a slice but it's switched to a `HashSet` for precisely this reason :)